rust/core-lib/src/editor.rs

```diff
@@ -712,6 +717,65 @@ impl Editor {
         */
     }
 
+    /// Indents or outdents lines based on selection and user's tab settings.
+    /// Uses a BTreeSet to holds the collection of lines to modify.
+    /// Preserves cursor position and current selection as much as possible.
+    /// Tries to have behavior consistent with other editors like Atom, Sublime and VSCode,
+    /// with non-caret selections not being modified.
+   
+    fn modify_indent(&mut self, direction: IndentDirection) {
+        let mut builder = delta::Builder::new(self.text.len());
+        let mut lines = BTreeSet::new();
+        let tab_text = if self.config.items.translate_tabs_to_spaces {
+                let tab_size = self.config.items.tab_size;
+                n_spaces(tab_size)
+            } else { "\t" };
+        for region in self.view.sel_regions() {
+            let (first_line, _) = self.view.offset_to_line_col(&self.text, region.min());
+            let (last_line, last_col) =
+                self.view.offset_to_line_col(&self.text, region.max());
+            let last_line = if last_col == 0 && last_line > first_line {
+                last_line - 1
+            } else {
+                last_line   
+            };
+            lines.insert(first_line);
+            lines.insert(last_line);
```

Comment:
This only inserts the first and last lines, where we want all the lines in between as well.